### PR TITLE
api-server: Api server with fastapi

### PR DESCRIPTION
**Pull request by koonpeng**
## What's new

same as #212, but with fastapi instead

## Self-checks

- [ ] I'm familiar with and follow this [ Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [ ] I added unit-tests for new components
- [ ] I tried testing edge cases
- [ ] I tested the behavior of the components that interact with the backend, with an e2e test

## Discussion

[Questions for reviewers]

### CODE REVIEW

File: packages/api-server/api_server/repositories/static_files.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        Parameters:
+            data:
+            path: relative path to save the file to
+        Returns:
+            a tuple containing the path the file is saved to and the url path that 
+            it is served from
+        '''
+        filepath = f'{self.directory}/{path}'
+        os.makedirs(os.path.dirname(filepath), exist_ok=True)
+        print(filepath)
```
Comment (matiasbavera) [inline packages/api-server/api_server/repositories/static_files.py:26]: Is this intentional?

**Review by matiasbavera — APPROVED** [review]
Non an expert but code looks good to me
